pkg/server/commands/listusers/list_users.go

```diff
@@ -13,64 +13,78 @@ type listUsersRequest interface {
 	GetRelation() string
 	GetUserFilters() []*openfgav1.ListUsersFilter
 	GetContextualTuples() *openfgav1.ContextualTupleKeys
+	GetDepth() uint32
 }
 
-type internalListUsersRequest struct {
+type ListUsersRequest struct {
```

Comment:
I'm changing this because it better aligns with the struct pattern we have in check – the protobuf request type and a second `ResolveCheckRequest`. This is also changed because this struct is no longer internal because we need the server to provide context on the depth limit.

Reply:
I'm uneasy about this package evolving in the way it is in a public package `pkg/commands/listusers`. This kind of code is internally facing code just to assist as helpers with the implementations of the RPC handlers in `pkg/server`. I would expect this to be a publicly exported type/struct in an internal package, but not a public package.

Though this isn't in the scope of this PR, unless we intend for this code and other `pkg/server/commands` code to be publicly consumed, then it really shouldn't be public code. We don't inten for any `pkg/server/commands` to be consumed, so it should all move to `internal` somewhere.

That being said, exporting this publicly in `pkg/server/commands/listusers` moves us even further in the wrong direct, so I'm not a huge fan of the change, but the issue is related to a bigger problem not in scope, so I'll just leave my opinion at that (no changes expected).

Reply:
What is actionable here? Admittedly, I'm not aware of all the context for why the code is organized the way it is, I'm mostly just trying to make it consistent with what we have w/ Check and ListObjects which is not exactly straightforward. 

For check, we introduce the internal `graph` package which contains some expansion logic, but just for check. For ListObjects we have `reverseexpand` which has its own set of exported types that may not be appropriate for export. Additionally, we have the commands abstraction which is not clear nor provides any structure. 

If there is something actionable, lets do it.

Reply:
> Additionally, we have the commands abstraction which is not clear nor provides any structure.
Agreed. The abstraction is just an extra level of indirection unnecessarily.

> If there is something actionable, lets do it.
Follow up action item (we need to create an issue for it) - move `pkg/server/commands` into `internal` as they are not intended for public consumption. As for this PR, no expected action items as this time. Will mark as resolved and track this as a separate issue.

Reply:
> because we need the server to provide context on the depth limit.

You can do this how ListObjects API does it

```go
     listUsersQuery := listusers.NewListUsersQuery(s.datastore, 
           WithResolveNodeLimit(s.resolveNodeLimit)
     return listUsersQuery.ListUsers(ctx, req)
```

and then you don't have to make the request object public.

Reply:
If it's of any help I agree with @jon-whit here around not making this an exported type. 

However I'd action this differently, I'd keep it private within this PR and in the following PR move the commands into the internal package and make them public. Otherwise users that directly import the cutting edge version of the openfga `pkg` package, might have a negative experience if they start relying on this and not having the context that this will be moved and made private again. 

The above scenario however unlikely, is a possibility and I'd advocate for us to keep building the muscle of minimizing any potential impact to our users.